sdk/core/System.ClientModel/src/Convenience/ClientCache.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A cache for storing client instances, ensuring efficient reuse.
+/// </summary>
+public class ClientCache
```

Comment:
Before we GA, we need to productize this type. In particular, we need to decide whether the 100 subclients fixed limit is fine and the cleanup logic is fine.

Reply:
Created an issue here - https://github.com/Azure/azure-sdk-for-net/issues/48278